xray/services/utils/xraywatchbody.go

```diff
@@ -0,0 +1,429 @@
+		}
+	case "":
+		// empty is fine
+	default:
+		return errors.New("Invalid Repository Type. Must be " + string(WatchRepositoriesAll) + " or " + string(WatchRepositoriesByName))
+	}
+
+	return nil
+}
+
+func createFilters(filters xrayWatchFilters, repo XrayWatchRepositoriesParams) []xrayWatchFilter {
+	result := []xrayWatchFilter{}
+
+	for _, packageType := range filters.PackageTypes {
+		filter := xrayWatchFilter{
+			Type:  "package-type",
+			Value: packageType,
+		}
+		result = append(result, filter)
+	}
+
+	for _, name := range filters.Names {
+		filter := xrayWatchFilter{
+			Type:  "regex",
+			Value: name,
+		}
+		result = append(result, filter)
+	}
+
+	for _, path := range filters.Paths {
+		filter := xrayWatchFilter{
+			Type:  "path-regex",
+			Value: path,
+		}
+		result = append(result, filter)
+	}
+
+	for _, mimeType := range filters.MimeTypes {
+		filter := xrayWatchFilter{
+			Type:  "mime-type",
+			Value: mimeType,
+		}
+		result = append(result, filter)
+	}
+
+	for key, value := range filters.Properties {
+		filter := xrayWatchFilter{
+			Type: "property",
+			Value: xrayWatchFilterPropertyValue{
+				Key:   key,
+				Value: value,
+			},
+		}
+		result = append(result, filter)
+	}
+
+	if repo.ExcludePatterns != nil || repo.IncludePatterns != nil {
+		filter := xrayWatchFilter{
+			Type: "path-ant-patterns",
+			Value: XrayWatchPathFilters{
+				ExcludePatterns: repo.ExcludePatterns,
+				IncludePatterns: repo.IncludePatterns,
+			},
+		}
+		result = append(result, filter)
+	}
+
+	return result
+}
+
+func configureBuilds(payloadBody *XrayWatchBody, params XrayWatchParams) error {
+	switch params.Builds.Type {
+	case WatchBuildAll:
+		allBuilds := xrayWatchProjectResourcesElement{
+			Name:     "All Builds",
+			Type:     "all-builds",
+			BinMgrID: params.Builds.All.BinMgrID,
+			Filters:  []xrayWatchFilter{},
+		}
+
+		if params.Builds.All.ExcludePatterns != nil || params.Builds.All.IncludePatterns != nil {
+			filters := []xrayWatchFilter{{
+				Type: "ant-patterns",
+				Value: XrayWatchPathFilters{
+					ExcludePatterns: params.Builds.All.ExcludePatterns,
+					IncludePatterns: params.Builds.All.IncludePatterns,
+				}},
+			}
+			allBuilds.Filters = filters
+		}
+
+		payloadBody.ProjectResources.Resources = append(payloadBody.ProjectResources.Resources, allBuilds)
+
+	case WatchBuildByName:
+		for _, byName := range params.Builds.ByNames {
+			build := xrayWatchProjectResourcesElement{
+				Type:     "build",
+				Name:     byName.Name,
+				BinMgrID: byName.BinMgrID,
+			}
+
+			payloadBody.ProjectResources.Resources = append(payloadBody.ProjectResources.Resources, build)
+		}
+	case "":
+		// empty is fine
+	default:
+		return errors.New("Invalid Build Type. Must be " + string(WatchBuildAll) + " or " + string(WatchBuildByName))
+	}
+
+	return nil
+}
+
+func configureBundles(payloadBody *XrayWatchBody, params XrayWatchParams) error {
```

Comment:
We prefer not to have placeholder code in the form of empty methods. I suggest you to the description of the ```CreateBody``` function a comment saying that bundles are currently not supported. This disclaimer should also be added to the API description in the README.